v3/pkg/application/linux_cgo.go

```diff
@@ -1570,11 +1571,12 @@ func runChooserDialog(window pointer, allowMultiple, createFolders, showHidden b
 					}
 					count++
 				}
-				close(selections)
-				C.gtk_widget_destroy((*C.GtkWidget)(unsafe.Pointer(fc)))
+			case C.GTK_RESPONSE_CANCEL:
+				// Cancelled
```

Comment:
Honestly I do not see any reason this needed to be switched from an `if statement` to a `switch statement` since the `cancel` case is ignored.  If you wanted it this way for consistency I could see the argument for that, but it appears to simply add extra lines of non-functional code IMHO.

Reply:
I was getting errors on my machine if cancel wasn't "handled"

Reply:
It's only an error in a `switch` statement.   My point is that if it was still an `if` statement then this wouldn't be an issue.   You could also just add a `default` case to handle that.  I still think this is more code than is necessary in this case since you can only have an `ok` or `not ok`.  Do you have any other changes you are waiting on?

I think I will go ahead and `approve / merge` this otherwise.

Reply:
repushed with swap back to if statement, should be good to ship on my end